client/chat/channel.tsx

```diff
@@ -412,82 +413,29 @@ export default function Channel(props: ChatChannelProps) {
     }
   }, [isLeavingChannel])
 
-  const [joinChannelError, setJoinChannelError] = useState<Error>()
-  const cancelJoinRef = useRef(new AbortController())
-
   useEffect(() => {
-    cancelJoinRef.current.abort()
-    const abortController = new AbortController()
-    cancelJoinRef.current = abortController
-
     if (isInChannel) {
-      dispatch(retrieveUserList(channelName))
-      dispatch(activateChannel(channelName) as any)
-    } else if (!isLeavingChannel) {
-      if (MULTI_CHANNEL) {
-        dispatch(
-          joinChannel(channelName, {
-            signal: abortController.signal,
-            onSuccess: () => setJoinChannelError(undefined),
-            onError: err => setJoinChannelError(err),
-          }),
-        )
-      } else {
-        push('/')
-      }
+      dispatch(retrieveUserList(channelId))
+      dispatch(activateChannel(channelId) as any)
```

Comment:
these casts can be removed now, should probably look through this PR and remove any other ones as well